test/testdrive/createdrop.td

```diff
@@ -226,3 +226,19 @@ a
   FORMAT BYTES;
 
 > DROP SOURCE f
+
+# This tests Materialize's logic around custom AWS regions
+! CREATE SOURCE custom_source
+  FROM KINESIS ARN 'arn:aws:kinesis:custom-region::stream/fake-stream'
+  WITH (access_key = 'fake_access_key', secret_access_key = 'fake_secret_access_key')
+  FORMAT BYTES;
+If providing a custom region, an `endpoint` option must also be provided
+
+> CREATE SOURCE custom_source
+  FROM KINESIS ARN 'arn:aws:kinesis:custom-region::stream/fake-stream'
+  WITH (access_key = 'fake_access_key',
+        secret_access_key = 'fake_secret_access_key',
+        endpoint = 'fake_endpoint')
+  FORMAT BYTES;
```

Comment:
Can you adapt this test into an example for the doc? I assume most folks who are actually building this will _get_ it, but seeing an example of the endpoint structure we expect would be nice. If not that, then maybe a link to the AWS docs to find the endpoint?